test/unit/property-base.test.js

```diff
@@ -24,11 +26,49 @@ describe('PropertyBase', function () {
                     _postman_two: 'something',
                     _three: _.noop
                 },
-                base = new PropertyBase(definition);
+                base = new Sdk.PropertyBase(definition);
 
             expect(base.meta('postman_one').postman_one).to.eql(definition._postman_one);
             expect(base.meta('postman_one')).to.not.have.property('postman_two');
             expect(base.meta('postman_one')).to.not.have.property('three');
         });
     });
+
+    describe('forEachParent helper', function () {
+        var expectedOrder = ['F2', 'F2.F3'],
+            collection = new Sdk.Collection(fixtures.nestedCollectionV2),
+            item = collection.items.members[1].items.members[0].items.members[0];
+
+        it('should set default options correctly when none are specified', function () {
+            var chain = [];
+
+            item.forEachParent(chain.unshift.bind(chain));
+            expect(_.map(chain, 'name')).to.eql(expectedOrder);
+        });
+
+        it('should return the collection as well when required', function () {
+            var chain = [];
+
+            item.forEachParent({ withRoot: true }, chain.unshift.bind(chain));
+            expect(chain.shift().id).to.be(fixtures.nestedCollectionV2.info._postman_id);
```

Comment:
This kinda thing is why you should use separate `expectedOrder` arrays to test. Needlessly combining the expected order has created this correct, but non-straightforward test.